scripts/quests/ahtUrhgan/Saga_of_the_Skyserpent.lua

```diff
@@ -0,0 +1,159 @@
+-----------------------------------
+-- Saga of the Skyserpent
+-- Log: 6, Quest: 43
+-- Fari-Wari: !pos 80 -6 -137 50
+-- qm7:       !pos -11 8 -185 62
+-- Biyaada:   !pos -65.802 -6.999 69.273 48
+-----------------------------------
+require('scripts/globals/common')
+require('scripts/globals/items')
+require('scripts/globals/quests')
+require('scripts/globals/interaction/quest')
+require('scripts/globals/npc_util')
+require('scripts/globals/titles')
+require('scripts/globals/zone')
+-----------------------------------
+
+local quest = Quest:new(xi.quest.log_id.AHT_URHGAN, xi.quest.id.ahtUrhgan.SAGA_OF_THE_SKYSERPENT)
+
+quest.reward =
+{
+    item = xi.items.IMPERIAL_GOLD_PIECE,
+    title = xi.title.SKYSERPENT_AGGRANDIZER
+}
+
+quest.sections =
+{
+
+    {
+        check = function(player, status, vars)
+            return status == QUEST_AVAILABLE
+        end,
+
+        [xi.zone.AHT_URHGAN_WHITEGATE] =
+        {
+            ['Fari-Wari'] = quest:progressEvent(823, { text_table = 0 }),
+
+            onEventFinish =
+            {
+                [823] = function(player, csid, option, npc)
+                    quest:begin(player)
+                end,
+            },
+        },
+    },
+
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED and vars.Prog == 0
+        end,
+
+        [xi.zone.AL_ZAHBI] =
+        {
+            ['Biyaada'] = quest:event(278),
+        },
+
+        [xi.zone.AHT_URHGAN_WHITEGATE] =
+        {
+            ['Fari-Wari'] = quest:event(829),
+        },
+
+        [xi.zone.HALVUNG] =
+        {
+            ['qm7'] =
+            {
+                onTrigger = function(player, npc)
+                    quest:setVar(player, 'Prog', 1)
+                    player:addKeyItem(xi.ki.LILAC_RIBBON)
+                    return quest:messageSpecial(zones[player:getZoneID()].text.KEYITEM_OBTAINED, xi.ki.LILAC_RIBBON)
+                end,
+            },
+        },
+    },
+
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED and vars.Prog == 1
+        end,
+
+        [xi.zone.AL_ZAHBI] =
+        {
+            ['Biyaada'] = quest:event(278),
+        },
+
+        [xi.zone.AHT_URHGAN_WHITEGATE] =
+        {
+            ['Fari-Wari'] = quest:progressEvent(953, { text_table = 0 }),
+
+            onEventFinish =
+            {
+                [953] = function(player, csid, option, npc)
+                    player:setPos(0, 0, 0, 0, xi.zone.WAJAOM_WOODLANDS)
+                end,
+            },
+        },
+
+        [xi.zone.WAJAOM_WOODLANDS] =
+        {
+            onZoneIn =
+            {
+                function(player, prevZone)
+                    if quest:getVar(player, 'Prog') == 1 then
+                        return 12
+                    end
+                end
+            },
+
+            onEventFinish =
+            {
+                [12] = function(player, csid, option, npc)
```

Comment:
While there's no problem with this, I'd be willing to bet retail does another onZoneIn here